js/app.js

```diff
@@ -48,10 +50,18 @@ $(document).ready(function() {
 
   // Input binding
   $searchInput
-  .on('keyup', function() {
-    var query = $(this).val();
-    toggleIconEmptyInput(query);
-    algoliaHelper.setQuery(query).search();
+  .on('keyup paste cut change', function() {
+    var $el = $(this);
+
+    // We use setTimeout to defer the callback to the bottom of the stack to
+    // catch the value after the paste, cut, or delete via mouse
+    setTimeout(function(){
```

Comment:
I would advocate the use of the event parameter to fetch the value instead of reaching for the DOM and waiting for it. For example in the case of a paste we could use : `arguments[0].originalEvent.clipboardData.getData('Text')`

Also we don't need to use the same callback for all the events we want to handle. In this case it would make sense, since we might want to use the event object that could have different form.